test/libsolidity/semanticTests/abiencodedecode/abi_encode_call1.sol

```diff
@@ -0,0 +1,75 @@
+pragma abicoder v2;
+contract C {
+	bool sideEffectRan = false;
+
+	function(uint256, string memory) external fPointer;
+
+	function f(uint256 p, string memory t) public {}
+
+	function test0() public returns (bool) {
+		return keccak256(f0()) == keccak256(f0Call());
+	}
+	function test1() public returns (bool) {
+		return keccak256(f1()) == keccak256(f1Call());
+	}
+	function test2() public returns (bool) {
+		return keccak256(f1Call()) == keccak256(f1s());
+	}
+	function test3() public returns (bool) {
+		return keccak256(fPointerCall()) == keccak256(f0());
+	}
+	function test4() public returns (bool) {
+		return keccak256(fLocalPointerCall()) == keccak256(f0());
+	}
+	function test5() public returns (bool) {
+		return keccak256(fReturnedFunctionPointer()) == keccak256(f0());
+	}
+
+	function f0() public pure returns (bytes memory) {
+		return abi.encodeWithSignature("f(uint256,string)", 1, "123");
+	}
+	function f0Call() public view returns (bytes memory) {
+		return abi.encodeCall(this.f, (1, "123"));
+	}
+	function f1() public pure returns (bytes memory) {
+		string memory x = "f(uint256,string)";
+		return abi.encodeWithSignature(x, 1, "123");
+	}
+	function f1Call() public view returns (bytes memory) {
+		return abi.encodeCall(this.f, (1,"123"));
+	}
+	string xstor;
+	function f1s() public returns (bytes memory) {
+		xstor = "f(uint256,string)";
+		return abi.encodeWithSignature(xstor, 1, "123");
+	}
+	function fPointerCall() public returns(bytes memory) {
+		fPointer = this.f;
+		return abi.encodeCall(fPointer, (1, "123"));
+	}
```

Comment:
Can you add a case where the pointer is a local variable or a parameter? This can actually matter in some cases (see e.g. #12057).

Reply:
Just a few lines below exactly that is done?

Reply:
Not sure how I missed that ;)